extensions/blocks/social-previews/editor.js

```diff
@@ -1,7 +1,31 @@
 /**
  * Internal dependencies
  */
-import { name, settings } from '.';
+import { name, settings, SocialPreviews } from '.';
 import registerJetpackPlugin from '../../shared/register-jetpack-plugin';
+import { registerPlugin } from '@wordpress/plugins';
+import getJetpackExtensionAvailability from '../../shared/get-jetpack-extension-availability';
 
+// Register the main "social-previews" extension if the feature is available
+// on the current plan.
 registerJetpackPlugin( name, settings );
+
+// TODO: remove in favour of `getJetpackExtensionAvailability()` call.
+// Testing purposes only to force reveal of upgrade nudge UI.
+const extensionAvailableOnPlan = getJetpackExtensionAvailability( 'social-previews' );
+// const extensionAvailableOnPlan = false;
+
+// If the social previews extension is **not** available on this plan (WP.com only)
+// then manually register a near identical Plugin which shows the upgrade nudge.
+// Note this is necessary because the official `registerJetpackPlugin` checks the
+// extension availability so will not render the Plugin if the extension is not
+// availabile.
+if ( !extensionAvailableOnPlan ) {
```

Comment:
```suggestion
if ( ! extensionAvailableOnPlan ) {
```

Reply:
I wonder why this one wasn't caught by the code formater. I'm pretty sure we use `!` the exact way @retrofox suggests. Maybe there isn't a rule?

Reply:
just in case, we can run `yarn reformat-files` and get some format suggestions. you'll get surprised about how many changes are applied by the script 😅

Reply:
That is odd...in fact my IDE isn't auto fixing the problem either.

Reply:
Well, the IDE is up to you and optional (I don't have anything there) but the precommit hook should format automatically, I think.